### PR TITLE
Core messaging implementation.

### PR DESCRIPTION
**Pull request by LMiljak**
-Added an extra module ContextMessages that both other two modules have as dependency (for it to work you have to run "mvn install" on the ContextProject)
-The module contains an abstract MessageListener class that should be extended in place of the JME3 MessageListener interface, because the MessageListener class only handles one specific message instead of multiple (which is bad practice).
-Accelerator (gyroscope) functionality has been moved to a separate class.

### CODE REVIEW

File: ContextApp/src/main/java/com/github/migi_1/ContextApp/AccelerometerSensor.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.util.Log;
+import java.util.concurrent.Callable;
+
+/**
+ *
+ * @author Marcel
+ */
+public class AccelerometerSensor extends Activity implements SensorEventListener {
```
Comment (NilsHullegien) [inline ContextApp/src/main/java/com/github/migi_1/ContextApp/AccelerometerSensor.java]: Missing class javadoc

File: ContextApp/src/main/java/com/github/migi_1/ContextApp/HelloActivity.java [hunk 2]
```diff
@@ -74,34 +74,39 @@ public void onCreate(Bundle savedInstanceState) {
...
+//
+//                //@Override
+//                public Object call() throws Exception {
+//                
+//                    /** Example of how you can make a call to a method in the game instance **/
+//                    application.gyroscopeChange();
+//                    return null;
+//                }
+//             });
+//        }
```
Comment (NilsHullegien) [inline ContextApp/src/main/java/com/github/migi_1/ContextApp/HelloActivity.java]: Please remove this.

File: ContextApp/src/main/java/com/github/migi_1/ContextApp/HelloActivity.java [hunk 3]
```diff
@@ -74,34 +74,39 @@ public void onCreate(Bundle savedInstanceState) {
...
 
         /** Auto-generated method **/
-        @Override
-        public void onAccuracyChanged(Sensor sensor, int i) {
+//        @Override
+//        public void onAccuracyChanged(Sensor sensor, int i) {
+//        }
+//
```
Comment (NilsHullegien) [inline ContextApp/src/main/java/com/github/migi_1/ContextApp/HelloActivity.java]: Please remove this.

**Comment by NilsHullegien** [comment]
It seems like there are still some PMD, and CheckStyle errors. 
Please fix these as well.

**Comment by LMiljak** [comment]
@NilsHullegien 
Appearently the ContextMessages module didn't even have the correct CheckStyle and pmd files in it. I've added those and fixed the errors they gave.
Also removed those useless comments pointed out by you.

**Comment by NilsHullegien** [comment]
Looks good now, thanks for adding the CheckStyle and pmd in the ContextMessages module as well.

**Comment by LMiljak** [comment]
Fixed merge conflict with master.

File: ContextApp/src/main/java/com/github/migi_1/ContextApp/AccelerometerSensor.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * Constructor for AccelerometerSensor.
+     * 
+     * @param act
+     *      The main activity from which is was created.
+     */
+    public AccelerometerSensor(HelloActivity act) {
+        this.act = act;
+    }
+
```
Comment (indisputable) [inline ContextApp/src/main/java/com/github/migi_1/ContextApp/AccelerometerSensor.java]: Please add javadoc

**Comment by dVoorhout** [comment]
everything else looks fine

**Pull request merged**